src/Strategy/Erlang/Rebar3Tree.hs

```diff
@@ -0,0 +1,156 @@
+module Strategy.Erlang.Rebar3Tree
+  ( discover
+  , analyze
+
+  , buildGraph
+  , rebar3TreeParser
+  , Rebar3Dep(..)
+  )
+  where
+
+import Prologue
+
+import Control.Carrier.Error.Either
+import qualified Data.Map.Strict as M
+import qualified Data.Text as T
+import Text.Megaparsec
+import Text.Megaparsec.Char
+
+import DepTypes
+import Discovery.Walk
+import Effect.Exec
+import Graphing (Graphing)
+import Effect.Grapher
+import Types
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files -> do
+  case find (\f -> fileName f `elem` ["rebar.config"]) files of
+    Nothing -> pure ()
+    Just _  -> runSimpleStrategy "erlang-rebar3tree" ErlangGroup $ analyze dir
+
+  pure WalkContinue
+
+rebar3TreeCmd :: Command
+rebar3TreeCmd = Command
+  { cmdNames = ["rebar3"]
+  , cmdBaseArgs = ["tree", "-v"]
+  , cmdAllowErr = Never
+  }
+
+analyze :: (Has Exec sig m, Has (Error ExecErr) sig m) => Path Rel Dir -> m ProjectClosureBody
+analyze dir = mkProjectClosure dir <$> execParser rebar3TreeParser dir rebar3TreeCmd []
+
+mkProjectClosure :: Path Rel Dir -> [Rebar3Dep] -> ProjectClosureBody
+mkProjectClosure dir deps = ProjectClosureBody
+  { bodyModuleDir    = dir
+  , bodyDependencies = dependencies
+  , bodyLicenses     = []
+  }
+  where
+  dependencies = ProjectDependencies
+    { dependenciesGraph    = buildGraph deps
+    , dependenciesOptimal  = NotOptimal
+    , dependenciesComplete = NotComplete
+    }
+
+type RebarGrapher = LabeledGrapher Rebar3Dep RebarLabel
+
+data RebarLabel =
+    RebarSource Text -- location
+  deriving (Eq, Ord, Show, Generic)
+
+buildGraph :: [Rebar3Dep] -> Graphing Dependency
+buildGraph deps = run . withLabeling toDependency $ do
+  traverse_ direct deps
+  traverse_ mkEdges deps
+
+  where
+  toDependency :: Rebar3Dep -> Set RebarLabel -> Dependency
+  toDependency pkg = foldr applyLabel start
+    where 
+    applyLabel :: RebarLabel -> Dependency -> Dependency
+    applyLabel (RebarSource src) dep = dep { dependencyLocations = src : dependencyLocations dep }
+
+    start =
+      Dependency { dependencyType =
+                      case T.isInfixOf (T.pack "github.com") (depLocation pkg) of
```

Comment:
[nit] `T.pack` on string literals is redunant because of `OverloadedStrings` -- so this can just be `T.isInfixOf "github.com" (depLocation pkg)`